src/Button/Button.js

```diff
@@ -118,15 +125,21 @@ export const Button = React.forwardRef(
         ref={ref}
         {...props}
       >
-        {children}
+        {iconPosition === 'right' ? <>{children}</> : null}
```

Comment:
Is there a reason children needs the fragment wrapper? Could we lose them so just 
{iconPosition === 'right' ? children : null}? 😄